SiraUtil/Plugin.cs

```diff
@@ -70,7 +72,9 @@ public void OnEnable()
         public void OnDisable()
         {
             _zenjectManager.Disable();
-            _harmony.UnpatchSelf();
+
+            // delay so DisposableManager is able to run before we unpatch on shutdown
+            UnityMainThreadTaskScheduler.Factory.StartNew(() => _harmony.UnpatchSelf()).ContinueWith((task) => Log.Error($"Failed to unpatch\n{task.Exception}"), TaskContinuationOptions.OnlyOnFaulted);
```

Comment:
This is kind of gross but I'm not sure what else to do. We need to delay unpatching when the game shuts down so all Dispose/LateDispose methods are called with the patch still active. I tried using `Application.quitting` but it unfortunately isn't set at this point. I got something working with the `Application.wantsToQuit` event but it was significantly more convoluted.